test/map_tool/test_segment_reshape_tool.py

```diff
@@ -179,6 +217,83 @@ def test_left_mouse_click_in_reshape_mode_adds_points_to_rubberbands(
     m_make_reshape_edits.assert_not_called()
 
 
+def test_pressing_backspace_in_reshape_mode_removes_point_from_rubberband(
+    map_tool: SegmentReshapeTool,
+):
+    map_tool.old_segment_rubber_band.setToGeometry(
+        QgsGeometry.fromWkt("LINESTRING(0 0, 1 1, 2 2, 3 3, 4 4, 5 5)")
+    )
+    map_tool.new_segment_rubber_band.setToGeometry(
+        QgsGeometry.fromWkt("LINESTRING(0 0, 1 1, 2 2, 3 3, 4 4, 5 5)")
+    )
+    map_tool.temporary_new_segment_rubber_band.setToGeometry(
+        QgsGeometry.fromWkt("LINESTRING(0 0, 1 1, 2 2, 3 3, 4 4, 5 5)")
+    )
+    map_tool.pick_location_mode = False
+    map_tool.reshape_mode = True
+
+    map_tool._handle_key_event(Qt.Key_Backspace)
+
+    assert map_tool.pick_location_mode is False
+    assert map_tool.reshape_mode is True
+    assert (
+        map_tool.new_segment_rubber_band.asGeometry().asWkt()
+        == "LineString (0 0, 1 1, 2 2, 3 3, 4 4)"
+    )
+    assert not map_tool.old_segment_rubber_band.asGeometry().isEmpty()
+    assert not map_tool.temporary_new_segment_rubber_band.asGeometry().isEmpty()
+
+
+def test_pressing_backspace_twice_in_reshape_mode_removes_points_from_rubberband(
+    map_tool: SegmentReshapeTool,
+):
+    map_tool.new_segment_rubber_band.setToGeometry(
+        QgsGeometry.fromWkt("LINESTRING(0 0, 1 1, 2 2, 3 3, 4 4, 5 5)")
+    )
+    map_tool.pick_location_mode = False
+    map_tool.reshape_mode = True
+
+    map_tool._handle_key_event(Qt.Key_Backspace)
+    map_tool._handle_key_event(Qt.Key_Backspace)
+
+    assert (
+        map_tool.new_segment_rubber_band.asGeometry().asWkt()
+        == "LineString (0 0, 1 1, 2 2, 3 3)"
+    )
+
+
+def test_pressing_backspace_in_reshape_mode_with_only_start_point_in_rubberband(
+    map_tool: SegmentReshapeTool,
+):
+    map_tool.new_segment_rubber_band.setToGeometry(
+        QgsGeometry.fromWkt("LINESTRING(0 0)")
+    )
+    map_tool.pick_location_mode = False
+    map_tool.reshape_mode = True
+
+    map_tool._handle_key_event(Qt.Key_Backspace)
+
+    assert map_tool.new_segment_rubber_band.asGeometry().asWkt() == "LineString (0 0)"
```

Comment:
this should probably change as well to reflect the previous comment, ie. backspace in this case should set the new segment to empty geometry